node/circuits.js

```diff
@@ -66,6 +73,9 @@ EndpointCircuits.prototype.getCircuit = function getCircuit(callerName, serviceN
             minRequests: stateOptions.minRequests,
             probation: stateOptions.probation
         });
+        circuit.stateChangedEvent.on(function circuitStateChanged(states) {
```

Comment:
Putting the logging and stats inline straight in `circuit` and skipping 3 event emissions is going to be faster.